### PR TITLE
Treat unknown OS like Linux

### PR DESCRIPTION
**Pull request by voidanix**
High chance it's an UNIX-like OS

Co-authored-by: Morel Bérenger

### CODE REVIEW

File: src/cmake/build_client.cmake [hunk 1]
```diff
@@ -25,6 +25,10 @@ if(BUILD_CLIENT)
         if(OSXCROSS_HOST)
             message(STATUS "osxcross build, building main.m for client")
             list(APPEND mac_client_sources xcode/main.m)
+        else()
```
Comment (TheAssassin) [inline src/cmake/build_client.cmake:28]: Why _exactly_ is this in the `APPLE` block? Does CMake consider every BSD to be macOS?

Comment (TheAssassin) [reply src/cmake/build_client.cmake:28]: You might also be able to check for *BSD by comparing `CMAKE_HOST_SYSTEM_NAME` against the string `BSD`.

Comment (voidanix) [reply src/cmake/build_client.cmake:28]: What about Solaris/Illumos or Haiku, though

Comment (TheAssassin) [reply src/cmake/build_client.cmake:28]: My main concern is why it is supposed to work if it's in an `elseif(APPLE)` block.

File: src/cmake/build_client.cmake [hunk 2]
```diff
@@ -26,8 +26,11 @@ if(BUILD_CLIENT)
             message(STATUS "osxcross build, building main.m for client")
             list(APPEND mac_client_sources xcode/main.m)
         endif()
-
         list(APPEND client_sources ${mac_client_sources})
+    elseif(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
```
Comment (TheAssassin) [inline src/cmake/build_client.cmake:30]: Can you please remove the first `if`? We have it twice in here now, but once should be enough with this `elseif` clause.

**Comment by TheAssassin** [comment]
Thanks a lot!

**Pull request merged**